File: LinearSearch.cpp
#include <stdio.h>
#include<conio.h>

int linearSearch(int arr[], int size, int key)
{
    if (size == 0) {
        return -1;
    }

    if (arr[size - 1] == key) {
        return size - 1;
    }
    return linearSearch(arr, size - 1, key);
}

int main()
{
    int arr[] = {45,63,33,98,56,23};
    int key = 56;
    int index
        = linearSearch(arr, sizeof(arr) / sizeof(int), key);
    if (index == -1) {
        printf("Key not found in the array.\n");
    }
    else {
        printf("The element %d is found at %d index of the "
               "given array \n",
               key, index);
    }
    return 0;
}

File: LinearSearch.java
public class LinearSearch {
    public static int Linear_search(int arr[],int key){
        for (int i=0;i<arr.length;i++){
            if (arr[i]==key){
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int arr[]={1,2,3,4,5,6,7,8,9,10};
        int key=8;
        int index=Linear_search(arr,key);
        if (index==-1){
            System.out.println("Element not found");
        }
        else{
            System.out.println("Element is found at index: "+index);
        }   
    }
}
